test/unit/storage_async_test.rb

```diff
@@ -0,0 +1,228 @@
+require File.expand_path(File.dirname(__FILE__) + '/../test_helper')
+
+class StorageAsyncTest < Test::Unit::TestCase
+  def test_basic_operations
+    storage = StorageAsync::Client.instance(true)
+    storage.del('foo')
+    assert_nil storage.get('foo')
+    storage.set('foo', 'bar')
+    assert_equal 'bar', storage.get('foo')
+  end
+
+  def test_redis_host_and_port
+    storage = StorageAsync::Client.send :new, url('127.0.0.1:6379')
+    assert_connection(storage)
+  end
+
+  def test_redis_url
+    storage = StorageAsync::Client.send :new, url('redis://127.0.0.1:6379/0')
+    assert_connection(storage)
+  end
+
+  def test_redis_unix
+    storage = StorageAsync::Client.send :new, url('unix:///tmp/redis_unix.6379.sock')
+    assert_connection(storage)
+  end
+
+  def test_redis_protected_url
+    assert_nothing_raised do
+      StorageAsync::Client.send :new, url('redis://user:passwd@127.0.0.1:6379/0')
```

Comment:
Why do we need this test?  Is it some internal method we call into?

Reply:
The test is to ensure URLs with user and password are accepted. This test is actually not much helpful right now because we don't support ACL in the `master` branch. But it's nice to have it though, as we'll add support for ACL and TLS in both sync and async modes soon.

Reply:
my point was that this is testing the underlying library. But ok, it makes sense to make sure this doesn't stop working silently.

Reply:
It's not testing the underlying library, it's testing `StorageAsync::Client`. Whether it uses another library internally to parse the URL it's not relevant for the test. This test is only to ensure `StorageAsync::Client` accepts URLs with username and passwords.